packages/DatePicker/src/DatePicker.js

```diff
@@ -82,18 +95,21 @@ function DatePicker(props) {
 
   function hideCalendar() {
     if (shouldShowCalendar) setShouldShowCalendar(false);
+    inputRef.current.blur();
   }
 
   function handleChange(newDate) {
     if (date !== newDate) onChange(newDate);
   }
 
   function handleClosePopover() {
-    window.requestAnimationFrame(() => {
+    setTimeout(() => {
       if (!isElementContainsFocus(calendarRef.current) && !isElementContainsFocus(inputRef.current)) {
+        setConfirmationResult(date ? moment.utc(date).format(humanFormat) : "");
+        setInputtedString(date ? moment.utc(date).format(dataFormat) : "");
         hideCalendar();
       }
-    });
+    }, 10);
```

Comment:
`requestAnimationFrame ` doesn't work anymore if I need to wait for newly added component to render. `setTimeout(, 0)` sometimes works, but not always, don't know what's the best way to do that. `setTimeOut(, 10)` will works better.

Reply:
When `setTimeout(, 0)` doesn't work, what does that look like?